pkg/providers/utils/manifest.go

```diff
@@ -166,12 +166,38 @@ func DiffPluginConfigs(olds, news []*apisixv1.PluginConfig) (added, updated, del
 	return
 }
 
+func DiffPluginMetadatas(olds, news []*apisixv1.PluginMetadata) (added, updated, deleted []*apisixv1.PluginMetadata) {
+	oldMap := make(map[string]*apisixv1.PluginMetadata, len(olds))
+	newMap := make(map[string]*apisixv1.PluginMetadata, len(news))
+	for _, pm := range olds {
+		oldMap[pm.Name] = pm
+	}
+	for _, pm := range news {
+		newMap[pm.Name] = pm
+	}
+
+	for _, pm := range news {
+		if ou, ok := oldMap[pm.Name]; !ok {
+			added = append(added, pm)
+		} else if !reflect.DeepEqual(ou, pm) {
```

Comment:
I think PluginMetadata is simple enough to compare values directly instead of using reflection.